src/datanode/src/instance.rs

```diff
@@ -82,6 +83,7 @@ pub struct Instance {
     pub(crate) sql_handler: SqlHandler,
     pub(crate) catalog_manager: CatalogManagerRef,
     pub(crate) table_id_provider: Option<TableIdProviderRef>,
+    row_inserter: Arc<RowInserter>,
```

Comment:
question: Why do we need to wrap `row_inserter` into an `Arc`? Looks like that `RowInserter` only holds a catalog manager ref, we could derive Clone for the RowInserter if we need to clone it.